### DISCUSSION TITLE
Compare with FluentAssertions

### DISCUSSION THREAD

**Discussion by MNF** [shouldly/shouldly#788] [Q&A] [answered]
I am using  FluentAssertions for some time and today noticed your library. From the quick look it has the same purpose and similar method names. According to  
[libhunt](https://dotnet.libhunt.com/compare-shouldly-vs-fluentassertions) FluentAssertions are a bit ahead of your library in popularity, but not essential. 

Can you provide the reasons why someone should choose Shouldly? What are the benefits ( and disadvantages) of using Shouldly compare to FluentAssertions ? 

It will be beneficial for everyone, if the answer for my question will be included in Readme.md or Wiki.

**Comment by josephwoodward** [comment, marked as answer]
I'm currently building a new site which will feature this information, but as someone that was using Shouldly and became a contributor the big sell for me and others is the simplicity of the assertions.

I've always felt the fluent API of FluentAssertions was unnecessary and added too much visual noise for me, for instance:

```csharp
var greeting = "Hello";
greeting.ShouldBe("Hello");
```
vs
```csharp
var greeting = "Hello";
greeting.Should().Be("Hello");
```

Shouldly (I believe) also had the friendly error messages first which has now made its way to Fluent Assertions.

Ultimately some people like one and others like another which is great, it's just down to preference. They're both great libraries and it's great there's more choice for everyone.
